src/app/config/thunks.js

```diff
@@ -143,53 +143,67 @@ export const approveCollectionRequest = (id, redirect) => dispatch => {
         });
 };
 
-export const loadTeamsRequest = () => dispatch => {
-    dispatch(actions.updateAllTeamsProgress());
-    teams
-        .getAll()
-        .then(response => {
-            if (!response) return dispatch(actions.updateAllTeamsFailure());
-            return dispatch(actions.updateAllTeams(response));
-        })
-        .catch(error => {
-            dispatch(actions.updateAllTeamsFailure());
-            switch (error.status) {
-                case 401: {
-                    // This is handled by the request function, so do nothing here
-                    break;
-                }
-                case "RESPONSE_ERR": {
-                    const notification = {
-                        type: "warning",
-                        message:
-                            "There's been a network error whilst trying to get teams. You may only be able to see previously loaded information and not be able to edit any team members.",
-                        isDismissable: true,
-                    };
-                    notifications.add(notification);
-                    break;
-                }
-                case "UNEXPECTED_ERR": {
-                    const notification = {
-                        type: "warning",
-                        message:
-                            "An unexpected error's occurred whilst trying to get teams. You may only be able to see previously loaded information and won't be able to edit any team members.",
-                        isDismissable: true,
-                    };
-                    notifications.add(notification);
-                    break;
+export const fetchGroupsRequest = isNewSignIn => dispatch => {
+    dispatch(actions.loadGroupsProgress());
+    isNewSignIn
+        ? teams
+            .getGroups()
+            .then(response => {
+                dispatch(actions.loadGroupsSuccess(response.groups));
+            })
+            .catch(error => {
+                dispatch(actions.loadGroupsFailure());
+                //TODO: map responses to user friendly by content designer
+                if (error) {
+                    notifications.add({ type: "warning", message: error?.message || error.status, autoDismiss: 5000 });
                 }
-                case "FETCH_ERR": {
-                    const notification = {
-                        type: "warning",
-                        message: "There's been a network error whilst trying to get teams. Try refresh the page.",
-                        isDismissable: true,
-                    };
-                    notifications.add(notification);
-                    break;
+                console.log(error);
+            })
+        : teams
+            .getAll()
+            .then(response => {
+                if (!response) return dispatch(actions.loadGroupsFailure());
```

Comment:
Not sure one line `if` statements have been used elsewhere in Florence, I think it is worth adding the braces around the block section even if it is more verbose. Makes it more readable and helps people who may be refactoring/modifying this section in the future not to accidentally miss the if statement.

Reply:
One line statement like doesn't need braces. It is valid javascript :-)